src/query/v2/requests.hpp

```diff
@@ -382,6 +389,7 @@ struct GetPropertiesRequest {
 
 struct GetPropertiesResponse {
   bool success;
+  std::optional<ShardError> error;
```

Comment:
Isn't the empty `error` mean the request was successful? If so, could you please remove the bool flags?